bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -202,122 +186,56 @@ public boolean loadClientPrice() {
+        Map<String, String> filters = new HashMap<>();
+        parameterMap.entrySet().forEach(filterCandidate -> {
+            String code = filterCandidate.getKey();
+            String[] value = filterCandidate.getValue();
 
-    @Nonnull
-    @Override
-    public Collection<ProductListItem> getProducts() {
-        Collection<ProductListItem> listItems = new ArrayList<>();
-
-        if (categoryRetriever != null && categoryRetriever.fetchCategory() != null) {
-            final CategoryProducts products = categoryRetriever.fetchCategory().getProducts();
-            if (products != null) {
-                for (ProductInterface product : products.getItems()) {
-                    try {
-                        boolean isStartPrice = product instanceof GroupedProduct;
-                        Price price = new PriceImpl(product.getPriceRange(), locale, isStartPrice);
-                        ProductImage smallImage = product.getSmallImage();
-                        listItems.add(new ProductListItemImpl(
-                            product.getSku(),
-                            product.getUrlKey(),
-                            product.getName(),
-                            price,
-                            smallImage == null ? null : smallImage.getUrl(),
-                            productPage,
-                            null,
-                            request,
-                            urlProvider));
-                    } catch (Exception e) {
-                        LOGGER.error("Failed to instantiate product " + product.getSku(), e);
-                    }
-                }
+            // we'll make sure there is a value defined for the key
+            if (value.length != 1) {
+                return;
             }
-        }
-        return listItems;
-    }
 
-    @Override
-    public AbstractCategoryRetriever getCategoryRetriever() {
-        return this.categoryRetriever;
+            filters.put(code, value[0]);
+        });
+
+        return filters;
     }
 
-    /**
-     * Obtains value from request for page, sets Pagination values for current, next and previous pages
-     *
-     * @return void
-     */
-    void setupPagination() {
-        int navPagesSize = 0;
-
-        navPagePrev = (navPageCursor <= 1) ? 1 : (navPageCursor - 1);
-        if ((getTotalCount() % navPageSize) == 0) {
-            navPagesSize = getTotalCount() / navPageSize;
-            navPageNext = (navPageCursor < (getTotalCount() / navPageSize)) ? (navPageCursor + 1) : navPageCursor;
-        } else {
-            navPagesSize = (getTotalCount() / navPageSize) + 1;
-            navPageNext = (navPageCursor < ((getTotalCount() / navPageSize) + 1)) ? (navPageCursor + 1) : navPageCursor;
+    @Override
+    public SearchResultsSet getSearchResultsSet() {
+        if (searchResultsSet == null) {
+            searchResultsSet = searchResultsService.performSearch(searchOptions, resource, productPage, request);
```

Comment:
Setting a query hook for the product query is supported via the 5th argument of the `performSearch` search method. However, currently a customer has no way of using it in any way using the Sling model delegation pattern. Replacing the whole query is also not possible.

The same applies for the `SearchResults` component.

Reply:
Right I missed that, I had this in my radar and then forgot to check it ... We definitely have to support this.